app/controllers/ReportController.java

```diff
@@ -162,24 +170,55 @@ public Result query(
      * Get specific report.
      *
      * @param id The identifier of the report.
-     * @return Matching report.
+     * @return The report, if any, otherwise notFound.
      */
-    public Result get(final String id) {
-        final Organization org;
+    public Result get(final UUID id) {
+        final Organization organization;
         try {
-            org = _organizationRepository.get(request());
+            organization = _organizationRepository.get(request());
         } catch (final NoSuchElementException e) {
-            return notFound();
+            return internalServerError();
+        }
+        try {
+            final Optional<Report> report = _reportRepository.getReport(id, organization);
+            return report.map(r -> ok(Json.toJson(models.view.reports.Report.fromInternal(r))))
+                    .orElseGet(ReportController::notFound);
+        } catch (final PersistenceException e) {
```

Comment:
At the moment the repositories throw runtime exceptions for unexpected failures which fall through the controllers and result in an automatic 5xx from Play.

Reply:
We really shouldn't be declaring or catching runtime exceptions.

https://docs.oracle.com/javaee/7/api/javax/persistence/PersistenceException.html

Reply:
See above.